src/main/java/net/glowstone/Explosion.java

```diff
@@ -327,13 +331,13 @@ private void playOutSoundAndParticles() {
         }
     }
 
-    private void playOutExplosion(GlowPlayer player, Iterable<BlockVector> blocks) {
+    private void playOutExplosion(GlowPlayer player, Iterable<Block> blocks) {
         Collection<Record> records = new ArrayList<>();
 
-        for (BlockVector block : blocks) {
-            byte x = (byte) (block.getBlockX() - location.getBlockX());
-            byte y = (byte) (block.getBlockY() - location.getBlockY());
-            byte z = (byte) (block.getBlockZ() - location.getBlockZ());
+        for (Block block : blocks) {
+            byte x = (byte) (block.getLocation().getBlockX() - (int) location.getX());
```

Comment:
Is there a reason to use `(int) location.getX()` instead of `location.getBlockX()`? Also, please cache the `block.getLocation()` object because the method creates new Location instances each time.